src/dbnode/storage/index/block_test.go

```diff
@@ -1885,16 +1846,19 @@ func TestBlockAggregateIterationErr(t *testing.T) {
 		iter.EXPECT().Current().Return([]byte("f1"), []byte("t1")),
 		iter.EXPECT().Next().Return(false),
 		iter.EXPECT().Err().Return(fmt.Errorf("unknown error")),
-		iter.EXPECT().Close().Return(nil),
 	)
 
 	ctx := context.NewBackground()
 	defer ctx.BlockingClose()
 
-	_, err = b.Aggregate(
+	aggIter, err := b.AggregateIter(ctx, results.AggregateResultsOptions())
+	require.NoError(t, err)
+	err = b.AggregateWithIter(
 		ctx,
+		aggIter,
 		QueryOptions{SeriesLimit: 3},
 		results,
+		10,
```

Comment:
how are we deciding on the limit here?

Reply:
pretty arbitrary. high enough for the tests to pass with a single iteration.